iphone/Maps/Core/iCloud/SynchronizationFileWriter.swift

```diff
@@ -253,20 +272,21 @@ final class SynchronizationFileWriter {
 
 // MARK: - URL + ResourceValues
 private extension URL {
-  func setResourceModificationDate(_ date: Date) throws {
+  func updateResourceModificationDate(_ date: Date = Date()) throws {
     var url = self
     var resource = try resourceValues(forKeys:[.contentModificationDateKey])
     resource.contentModificationDate = date
     try url.setResourceValues(resource)
   }
 }
 
+// MARK: - Date + WritingWithUpdateResourceModificationDate
+
 private extension Data {
   func write(to url: URL, options: Data.WritingOptions = .atomic, lastModificationDate: TimeInterval? = nil) throws {
     try write(to: url, options: options)
     if let lastModificationDate {
-      try url.setResourceModificationDate(Date(timeIntervalSince1970: lastModificationDate))
+      try url.updateResourceModificationDate(Date(timeIntervalSince1970: lastModificationDate))
```

Comment:
Why is this call needed?

Reply:
When we create a file in the cloud its `lastModified` date will be equal to the `creationDate` and will **not** be equal to the source local file's `lastModified`. 
In this case, we will have situation:
1. local file with the lastModified date 0001
2. copy the local file to the cloud container
3. cloud file (the same as local) last modified date is 0003 (now)
4. difference in the times will trigger the local file updation

It is important to keep the `lastModified`  time the same as the default c++ `move` do (FileManager behavior is different).

Reply:
Won't the file updation be triggered anyway before the modification time is reset to the original file's time?

Reply:
I've tested this behaviour - no, this update will be triggered only once because the writing operation is synchronous in the fileCoordinator's scope.